src/IterableCodeExtractor.php

```diff
@@ -262,6 +262,23 @@ static function ( $file, $key, $iterator ) use ( $include, $exclude, $extensions
 		return $filtered_files;
 	}
 
+	private static function file_has_file_extension( $file, $extensions ) {
+		return in_array( $file->getExtension(), $extensions, true ) ||
+			in_array( static::file_get_extension_multi( $file ), $extensions, true );
+	}
+
+	private static function file_get_extension_multi( $file ) {
+		$file_extension_separator = '.';
+
+		$filename = $file->getFilename();
+		$parts    = explode( $file_extension_separator, $filename, 2 );
+		if ( count( $parts ) <= 1 ) {
+			// if ever something goes wrong, fall back to SPL
+			return $file->getExtension();
+		}
+		return $parts[1];
+	}
```

Comment:
PHPDoc blocks for these two methods and perhaps some more explicit test coverage in `IterableCodeExtractorTest` would be nice, so it's more obvious what they do.